src/xrc/src/api.rs

```diff
@@ -356,19 +357,31 @@ async fn handle_cryptocurrency_pair(
     }
 
     if !utils::is_caller_privileged(&caller) {
+        let current_timestamp = env.time_secs();
+        let timestamp_is_in_future = timestamp.value > current_timestamp;
```

Comment:
Does it make sense to allow a privileged canister to request a timestamp in the future?
If it results in simpler code, then the answer is probably "yes"...

Reply:
The code is simpler, but it would not be much too much work. When I work on the duplicate code, I'll address this as well.